src/loaders.rs

```diff
@@ -78,16 +109,24 @@ impl FileSystemLoader {
     }
 }
 
-pub struct AppDirsLoader {}
+pub struct AppDirsLoader {
+    encoding: &'static Encoding,
+}
 
 impl AppDirsLoader {
+    pub fn new(encoding: &'static Encoding) -> Self {
+        Self { encoding }
+    }
+
     fn get_template(
         &self,
-        _py: Python<'_>,
-        _template_name: &str,
-        _engine: &EngineData,
+        py: Python<'_>,
+        template_name: &str,
+        engine: &EngineData,
     ) -> Result<PyResult<Template>, LoaderError> {
-        todo!()
+        let dirs = get_app_template_dirs(py, "templates").unwrap_or_default();
```

Comment:
I think it's more correct to raise here than to swallow errors:

```suggestion
        let dirs = match get_app_template_dirs(py, "templates") {
            Ok(dirs) => dirs,
            Err(e) => return Ok(Err(e))
        };
```

Note that we can't use `?` because we want to return a `PyErr`, not a `LoaderError`.